Build failure: python3Packages.torch on x86_64-darwin

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

On an Intel Mac:

```console
$ nix build 'nixpkgs#python3Packages.torch'
$ # OR
$ nix-build '<nixpkgs>' -A python3Packages.torch
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/304106458

### Relevant log output

```console
[…clipped to just the OpenMP stuff at the end - see the Hydra link for the full log…]
CMake Warning (dev) at /nix/store/y50arqd9pz1f3cyr1rafp176wxig8ws5-cmake-3.31.7/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
  The package name passed to `find_package_handle_standard_args` (OpenMP_C)
  does not match the name of the calling package (OpenMP).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Modules/FindOpenMP.cmake:623 (find_package_handle_standard_args)
  third_party/ideep/mkl-dnn/cmake/OpenMP.cmake:41 (find_package)
  third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
CMake Warning (dev) at /nix/store/y50arqd9pz1f3cyr1rafp176wxig8ws5-cmake-3.31.7/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
  The package name passed to `find_package_handle_standard_args` (OpenMP_CXX)
  does not match the name of the calling package (OpenMP).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Modules/FindOpenMP.cmake:623 (find_package_handle_standard_args)
  third_party/ideep/mkl-dnn/cmake/OpenMP.cmake:41 (find_package)
  third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
CMake Error at third_party/ideep/mkl-dnn/cmake/OpenMP.cmake:88 (message):
  OpenMP library could not be found.  Proceeding might lead to highly
  sub-optimal performance.
Call Stack (most recent call first):
  third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)


-- Configuring incomplete, errors occurred!
```

### Additional context

Saw this one near the top of [zh.fail](https://zh.fail). Seems to have been caused by #424873, which removed `openmp` from `torch`'s `buildInputs`. That may have fixed ARM64 builds, but it's causing Intel builds to fail to find OpenMP for some reason.

```console
$ # Fails with the last commit of #424873:
$ nix build 'github:NixOS/nixpkgs/85038ff6aeb8fee76530ade86e060e5ca0307ccb#python3Packages.torch'
warning: download buffer is full; consider increasing the 'download-buffer-size' setting
error: builder for '/nix/store/xfqxc93qbiz30gi4z0gd5jipamld5wp3-python3.13-torch-2.7.1.drv' failed with exit code 1;
       last 25 log lines:
       >   third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)
       > This warning is for project developers.  Use -Wno-dev to suppress it.
       >
       > -- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
       > CMake Warning (dev) at /nix/store/y50arqd9pz1f3cyr1rafp176wxig8ws5-cmake-3.31.7/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
       >   The package name passed to `find_package_handle_standard_args` (OpenMP_CXX)
       >   does not match the name of the calling package (OpenMP).  This can lead to
       >   problems in calling code that expects `find_package` result variables
       >   (e.g., `_FOUND`) to follow a certain pattern.
       > Call Stack (most recent call first):
       >   cmake/Modules/FindOpenMP.cmake:623 (find_package_handle_standard_args)
       >   third_party/ideep/mkl-dnn/cmake/OpenMP.cmake:41 (find_package)
       >   third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)
       > This warning is for project developers.  Use -Wno-dev to suppress it.
       >
       > -- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
       > -- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND)
       > CMake Error at third_party/ideep/mkl-dnn/cmake/OpenMP.cmake:88 (message):
       >   OpenMP library could not be found.  Proceeding might lead to highly
       >   sub-optimal performance.
       > Call Stack (most recent call first):
       >   third_party/ideep/mkl-dnn/CMakeLists.txt:89 (include)
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix-store -l /nix/store/xfqxc93qbiz30gi4z0gd5jipamld5wp3-python3.13-torch-2.7.1.drv
$ # But works with the commit that PR was based on:
$ nix build 'github:NixOS/nixpkgs/669d470a1c2d4b9192f75626e2b997f316c10af8#python3Packages.torch'
[no output - successfully fetches cached version]
```

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixpkgs-25.05-darwin"`
 - nixpkgs: `/nix/store/j9k79bd6m5ryxc6kcmii07r41751zhz2-nixpkgs-25.05-darwin/nixpkgs`
   - (Note: That channel is 25.05, which doesn't have the issue, but I was actually testing the build on unstable using the flake.)

### Notify maintainers

cc @GaetanLepage @teh @thoughtpolice @tscholak

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/torch/source/default.nix
+++ b/pkgs/development/python-modules/torch/source/default.nix
@@ -526,9 +526,9 @@ buildPythonPackage rec {
     blas
     blas.provider
   ]
-  # Including openmp leads to two copies being used. This segfaults on ARM.
+  # Including openmp leads to two copies being used on ARM, which segfaults.
   # https://github.com/pytorch/pytorch/issues/149201#issuecomment-2776842320
-  # ++ lib.optionals stdenv.cc.isClang [ llvmPackages.openmp ]
+  ++ lib.optionals (stdenv.cc.isClang && !stdenv.hostPlatform.isAarch64) [ llvmPackages.openmp ]
   ++ lib.optionals cudaSupport (
     with cudaPackages;
     [

